### DISCUSSION TITLE
Custom Doc Templates

### DISCUSSION THREAD

**Discussion by hkkmalinda** [toeverything/AFFiNE#6178] [Ideas]
### Description

Hi, 
It would be really nice if you could make AFFiNe automatically use a predefined template (by the user) for every doc when it was created. Also, it is better to create a template creation method too.

### Use case

Let's say for a journal doc I like to use a specific table so when creating a doc I need that table to be added on every doc. So if I can use a doc template with the creation of the doc that will be very easy. 

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

[a comment by an automated account is not shown]

**Comment by hkkmalinda** [comment]
I think this feature is a good one to have in the app

**Comment by Arcadianer** [comment]
I agree. I use templates in other notes app all the time to provide a good structure in all of my blocks. Combine this with the new property system and you have a really powerful tool. 
This is also one of the showstopper points for me that prohibit me from fully switching over to this tool for my everyday work.

**Comment by bigint** [comment]
+1 for this idea!

**Comment by davidsmorais** [comment]
+1 for this 

Would also add a 'Custom Journal' template

**Comment by vikram-j-s** [comment]
Yes. I want to be able to create templates and use public templates created by others. Need more choices in the template library.

**Comment by houshasen** [comment]
Please include "property" as part of the template as well. This means listing the properties, not their values. This step is crucial for making object note-taking more realistic.

https://github.com/toeverything/AFFiNE/issues/6922

**Reply by lexfreedomcom** [reply]
Hi! It would be great if you add page templates.

Story: 
As user I want to add page template so that I can use it when creating new page from template.
 - add text in template (structure with headers)
 
 As user I want to create new page from template so that I won't spend time for writing the same things.
 - from / mode page is adding from template 
 - can be created from +
 - seperate screen with all template , can create from there

**Comment by jnachman123** [comment]
+ 1 for me , this is a really great feature in obsidian
